### DISCUSSION TITLE
Using Manga Tagger for Existing Manga

### DISCUSSION THREAD

**Discussion by exedox** [Inpacchi/Manga-Tagger#8] [Q&A] [answered]
I asked this in the reddit post, but I figured if it's here, then other people can at least see the answer easy peasy. So I'm curious if Manga-Tagger can also be run on pre-existing manga downloaded via FMD2.  Also thanks so much for developing this!  It's always nice to see people provide some attention to manga/anime in this way.

Also, say if something goes down and the source of a manga is no longer available and chapters need to be continued via another source, will Manga-Tagger have any issues with the change or will it still register the same manga via title/author and continue tagging as it should?

**Comment by Inpacchi** [comment, marked as answer]
I just responded to ya, but I'll also throw my response here 🌮 

Currently, Manga Tagger **cannot** be run on existing manga. The reason for this is because when the chapters are downloaded, Manga Tagger sets delimiters in FMD2 so that the filenames can be parsed without (much) error. 

However! I have been planning to develop CLI functionality so that Manga Tagger can be manually run; however there will need to be standards for how the files are named so Manga Tagger can correctly parse the series and chapter numbers.

How do you have your manga folders structured? Mines are currently <SERIES_TITLE>/<SERIES_TITLE CHAPTER_NUMBER.cbz>, so `One Piece/One Piece 001.cbz` is an example.

Manga Tagger is completely independent of source. The only thing that Manga Tagger parses for is the manga title and chapter number, so it doesn't matter what source the manga is downloaded from.

**Reply by exedox** [reply]
Thanks so much for the response! I was hoping to do <SERIES_TITLE>/<SERIES_TITLE - Chapter CHAPTER_NUMBER.cbz> and going off your example One Piece/One Piece - Chapter 001.cbz 

The chapters currently are going Chapter 1, but I plan to rename them.  I haven't created a huge library just yet because I was figuring out how to tag and sort them properly.  Which is why I'm glad you came up with this when you did, because then now I don't have to worry too much about redoing everything.  I'll just have this set up on my desktop from scratch (currently using FMD2 on my laptop.).  I'm assuming the structure of the manga will need to be set up through FMD2, should I do anything extra to make sure manga tagger does the same thing?

**Reply by Inpacchi** [reply]
If you follow the [instructions here](https://github.com/Inpacchi/Manga-Tagger/wiki/Installation-&-Configuration), you'll have the same exact setup that I do currently.

Manga Library at C:\Library, all downloads at C:\Downloads, with the library structure as follows:
```
SERIES_TITLE
  | SERIES_TITLE CHAPTER_NUMBER.cbz

One Piece
  | One Piece 001.cbz
```

If you would prefer `One Piece Chapter 001.cbz` or `One Piece - Chapter 001.cbz` I can make a change in the code to allow for custom renaming. Which I guess I should do anyways, lol.

**Reply by exedox** [reply]
Yes please!!  I really appreciate you taking all this into consideration.  It's not too high of a priority because everything is still functional. :3 you made this happen so I'm pretty thankful for it.

**Comment by exedox** [comment]
Inpacchi, let me know when you impliment the change in code please.  :3 there's no rush. I have been holding off installing manga tagger until the new change has been made, but decided maybe it's better leave it as your set up because it's not too different from how I wanted to set up the library structure.   (I check beck every now and then to see if there's new updates).

I'm going to install Manga Tagger as is for now and will probably follow up with more questions later.

**Reply by Inpacchi** [reply]
No problem! I've been working on fleshing out a few things and fixing a couple bugs before I get to the CLI implementation, but if you have any questions please feel free to ask away :)

**Reply by exedox** [reply]
haha so I did a thing where I wanted to store all my manga in a different directory but I think I messed it up.  I wanted everything stored on my larger I: drive and it kinda didn't work out as I planned :(.  Could you also make it so we can have the library located in a different location? xD  Is there a way to undo what I did?

I tried to redo everything following all the steps but nothing is found in the C:Library dir.

I get the error manga_tagger: Unexpected status SERVICE_PAUSED in response to START control.

I'm going to start an issue as I ran into this problem again on another machine.

EDIT: For later, I am wondering if it's possible to add volumes to the tagger options. Just as anime has seasons, manga has volumes.  However, I noticed the databases you're using don't have the chapters listed with their corresponding volumes, so that might be difficult. I'll reach out to you about it later.  Typing this now to remind myself.
